### ISSUE TITLE
Opera GX - Crashing/Closing Browser on Startup.

### ISSUE THREAD

**Issue by Jirodyne** [sienori/Tab-Session-Manager#1333]
## Short description

When I click on the browser icon to start up Opera GX, the browser opens for a second, then closes the session to open the Tab Session Manager's "Last Session", only to immediately close/crash instead.

## Steps to reproduce

1. Start browser
2. Turn on the setting: Behavior when starting the browser - Open previous session
3. Close Browser
4. Start Browser and watch it close/crash before fully loading up

## Expected result

To load up the Session I closed the browser with.

## Actual result

It loads up the default browser, but when it closes it to load the session from before, it closes/crashes the browser.

I will edit/post the Error Log after posting this. didn't think to get it before righting this and don't want to cause the crash again and lose all of this

## Platform information
* Platform (OS):  Windows 10
* Browser: Opera GX
* Version of browser:  LVL 5 (core: 109.0.5097.100)
* Version of Tab Session Manager: 7.0.0

## Additional context

I can get around the issue, by either Removing Tab sessions Manage (Don't want to do that.). Open a Private Tab first by right clicking the browser and then manually loading the session saved (That's a hassle and don't want to be forced to do that every time), or turning off the "Behavior when starting the browser - Open previous session" to "Do Nothing" option, which I have used right now, switching my browser's settings to save the session and not start with an empty page.

This was never an issue before today. I don't know what change, there was no open on Opera GX, no windows update, as far as I know no Tab Session Manage Update last night while I was sleeping and PC off. No idea what changed to cause this.

**Comment by cocoonkid** [comment]
Same Issue here on Linux. 

![image](https://github.com/sienori/Tab-Session-Manager/assets/43231640/076d9b6f-2c84-4254-b731-d58d1127bc27)


I can confirm this happens as soon as you select thet option to reopen last sessions tabs OR the option to save them to the Downloads folder.

**Comment by slovia** [comment]
Exact same problem here in Floorp on Linux. To me it looks like it started with the latest update...

**Comment by eylenburg** [comment]
Same here on Firefox 126

**Comment by TheJRC** [comment]
Got the same issue on Firefox 126.0.1, although I'm unsure if Tab Session Manager had been updated yet when the error occurred. Due to the behavior, I assumed it was the problem, opened my browser in Safe Mode, and updated extensions. After a bit more weirdness and another crash, it seems to be working again now, though I don't wish to close my browser again to test it out.

**Comment by modball** [comment]
> it seems to be working again now, though I don't wish to close my browser again to test it out.

I was able to my browser to launch successfully with v7 by repeatedly retrying. 5 or 6 attempts would usually do it. 

For now probably best to revert to 6.12.2 and turn off auto updates temporarily.

**Comment by TheJRC** [comment]
> For now probably best to revert to 6.12.2 and turn off auto updates temporarily.

I am unsure how to revert. I was thinking of setting startup behavior to "Do Nothing" and seeing if manually loading session will work, but if not, then yeah, reverting would work.

**Comment by modball** [comment]
> I am unsure how to revert.

You can delete Tab-Session-Manager@sienori.xpi. from the extensions folder in your user profile folder. Something similar to:

C:\Users\%your-user-name%\AppData\Roaming\Mozilla\Firefox\Profiles\%random-characters%.default\extensions\

Then download/install version 6.12.2 from here: https://addons.mozilla.org/en-CA/firefox/addon/tab-session-manager/versions/

Then open Tools, Add-ons and Themes, use the small **...** menu for Tab Session Manager to choose Manage and then turn off auto updating.

**Comment by TheJRC** [comment]
Oh, didn't realize there's a Versions page. Wouldn't just uninstalling it normally work? Is hunting down the XPI and manually deleting it necessary?

**Comment by modball** [comment]
> Wouldn't just uninstalling it normally work?

Yes that would work if you are able to get your browser to launch successfully. Mine was crashing on load about 95% of the time.

Best to back up your session(s) just in case.

**Comment by tkhyn** [comment]
Same issue here on Firefox 126.0.1, on multiple machines and OSes (Windows 11, Linux Mint). Reverting to 6.* and disabling auto-updates fixes the problem.

**Comment by sienori** [comment]
Thank you for the report!
I have discovered an issue where the window may close before the session opens if the startup behavior is set to 'Open previous session.' 
 This issue will be fixed in version 7.0.1.

Please note that we do not support operation on browsers other than Firefox, Chrome, and Edge, or if the browser version is outdated.

**Comment by dkangel666** [comment]
Can someone explain to me how to update to version 7.0.1 from the zip file ?

A big thank you

**Comment by nerdCopter** [comment]
same behavior, ans confirming i have `Open previous session` enabled.

**Comment by andrhua** [comment]
to disable extension for the time being, run firefox in a safemode (`firefox --safe-mode`) and disable tsm from there
